### PR TITLE
RGB Matrix Caps Lock and Layer indicator example

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

Provide some examples on using RGB Matrix as indicators, a frequently asked question on the QMK discord.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

File: docs/feature_rgb_matrix.md [hunk 1]
```diff
@@ -624,6 +624,37 @@ void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
 }
 ```
 
+### Indicator Examples :id=indicator-examples
+
+Caps Lock indicator on alpha numberic flagged keys:
```
Comment (fauxpark) [inline docs/feature_rgb_matrix.md:629]:
```suggestion
Caps Lock indicator on alphanumeric flagged keys:
```

**Review by drashna — COMMENTED** [review]
Would prefer that both of these use the "advanced" function examples, as they're broken up into chunks for faster rendering

**Comment by filterpaper** [comment]
> Would prefer that both of these use the "advanced" function examples, as they're broken up into chunks for faster rendering

I've updated the function example as advised.

File: docs/feature_rgb_matrix.md [hunk 2]
```diff
@@ -624,6 +624,37 @@ void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
...
+
+Caps Lock indicator on alphanumeric flagged keys:
+```c
+void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
+    if (host_keyboard_led_state().caps_lock) {
+        for (uint8_t i = 0; i < DRIVER_LED_TOTAL; ++i) {
+            if (g_led_config.flags[i] & LED_FLAG_KEYLIGHT) {
+                RGB_MATRIX_INDICATOR_SET_COLOR(i, RGB_RED);
+            }
+        }
```
Comment (drashna) [inline docs/feature_rgb_matrix.md:637]: To make things further simplified
```suggestion
        for (uint8_t i = led_min; i <= led_max; i++) {
            if (g_led_config.flags[i] & LED_FLAG_KEYLIGHT) {
                rgb_matrix_set_color(i, RGB_RED);
            }
        }
```

File: docs/feature_rgb_matrix.md [hunk 3]
```diff
@@ -624,6 +624,37 @@ void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
...
+    switch(get_highest_layer(layer_state|default_layer_state)) {
+        case RAISE:
+            rgb_matrix_set_color_all(RGB_BLUE);
+            break;
+        case LOWER:
+            rgb_matrix_set_color_all(RGB_YELLOW);
+            break;
+        default:
+            break;
+    }
```
Comment (drashna) [inline docs/feature_rgb_matrix.md:654]:
```suggestion
    for (uint8_t i = led_min; i <= led_max; i++) {
        switch(get_highest_layer(layer_state|default_layer_state)) {
            case RAISE:
                rgb_matrix_set_color(i, RGB_BLUE);
                break;
            case LOWER:
                rgb_matrix_set_color(i, RGB_YELLOW);
                break;
            default:
                break;
        }
    }
```

**Review by drashna — COMMENTED** [review]
This may be better, given how the functions work, and what you're trying to do.

**Review by drashna — APPROVED** [review]
Awesome, thanks!

**Pull request merged**